types/inquirer/index.d.ts

```diff
@@ -10,430 +10,909 @@
+        type PromptOptions<T extends Question = Question> = T & {
             /**
-             * Called when the UI closes. Override to do any specific cleanup necessary
+             * The choices of the prompt.
              */
-            close(): void;
-            /**
-             * Generate the prompt question string
-             */
-            getQuestion(): string;
-        }
-    }
+            choices: Choices;
+        };
 
-    namespace ui {
+        /**
+         * Represents the state of a prompt.
+         */
+        type PromptState = LiteralUnion<"pending" | "idle" | "loading" | "answered" | "done">;
 
         /**
-         * Base interface class other can inherits from
+         * Represents a prompt.
          */
-        interface BaseUI {
-            rl: ReadlineInterface;
-            new(opt: StreamOptions): BaseUI;
+        interface PromptBase {
             /**
-             * Handle the ^C exit
-             * @return {null}
+             * Gets or sets a string which represents the state of the prompt.
              */
-            onForceClose(): void;
+            status: PromptState;
+
             /**
-             * Close the interface and cleanup listeners
+             * Runs the prompt.
+             *
+             * @returns
+             * The result of the prompt.
              */
-            close(): void;
+            run(): Promise<any>;
         }
+
         /**
-         * Base interface class other can inherits from
+         * Provides the functionality to initialize new prompts.
          */
-        interface PromptUI extends BaseUI {
-            process: Observable<Answer>;
-            new(prompts: Prompts, opt: StreamOptions): PromptUI;
-            run<A>(questions: Questions<A>): Promise<A>;
+        interface PromptConstructor {
             /**
-             * Once all prompt are over
+             * Initializes a new instance of a prompt.
+             *
+             * @param question
+             * The question to prompt.
+             *
+             * @param readLine
+             * An object for reading from the command-line.
+             *
+             * @param answers
+             * The answers provided by the user.
              */
-            onCompletion(): void;
-            processQuestion<A>(question: Question<A>): Observable<Question<A>>;
-            fetchAnswer<A>(question: Question<A>): Observable<Question<A>>;
-            setDefaultType<A>(question: Question<A>): Observable<Question<A>>;
-            filterIfRunnable<A>(question: Question<A>): Observable<Question<A>>;
+            new(question: any, readLine: ReadlineInterface, answers: Answers): PromptBase;
         }
 
         /**
-         * Sticky bottom bar user interface
+         * Provides a set of prompt-constructors.
          */
-        interface BottomBar extends BaseUI {
-            new(opt?: StreamOptions & { bottomBar?: string }): BottomBar;
+        interface PromptCollection {
```

Comment:
```suggestion
        type PromptCollection = Record<string, PromptConstructor>;
```
You would also need to delete some lines below.

Reply:
Thanks for the suggestion - I'm going to commit this instantly.